src/site/content/en/blog/bundling-non-js-resources/index.md

```diff
@@ -0,0 +1,193 @@
+
+```shell
+$ emcc input.cpp -o output.mjs
+## or, if you don't want to use .mjs extension
+$ emcc input.cpp -o output.js -s EXPORT_ES6
+```
+
+When using this option, the output will use the `new URL(..., import.meta.url)` pattern under the hood, so that bundlers can find the associated Wasm file automatically.
+
+You can also use this option with [WebAssembly threads](https://web.dev/webassembly-threads/#c) by adding a `-pthread` flag:
+
+```shell
+$ emcc input.cpp -o output.mjs -pthread
+## or, if you don't want to use .mjs extension
+$ emcc input.cpp -o output.js -s EXPORT_ES6 -pthread
+```
+
+In this case, the generated Web Worker will be included in the same fashion and will also be discoverable by bundlers and browsers alike.
+
+##### Rust via wasm-pack / wasm-bindgen
+
+[wasm-pack](https://github.com/rustwasm/wasm-pack)—the primary Rust toolchain for WebAssembly—also has several output modes.
+
+By default, it will emit a JavaScript module that relies on the [WebAssembly ESM integration proposal](https://github.com/WebAssembly/esm-integration). At the moment of writing, this proposal is still experimental, and the output will work only when bundled with Webpack.
+
+Instead, you can ask wasm-pack to emit a browser-compatible ES6 module via `--target web`:
+
+```shell
+$ wasm-pack build –target web
+```
+
+The output will use the described `new URL(..., import.meta.url)` pattern, and the Wasm file will be automatically discovered by bundlers as well.
+
+If you want to use WebAssembly threads with Rust, the story is a bit more complicated. Check out the [corresponding section of the guide](https://web.dev/webassembly-threads/#rust) to learn more.
+
+Short version is that you can't use arbitrary thread APIs, but if you use [Rayon](https://github.com/rayon-rs/rayon), you can combine it with the [wasm-bindgen-rayon](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon) adapter so that it can spawn Workers on the Web. The JavaScript glue used by wasm-bindgen-rayon [also includes](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/blob/4cd0666d2089886d6e8731de2371e7210f848c5d/demo/index.js#L26) the `new URL(...)` pattern under the hood, and so the Workers will be discoverable and included by bundlers as well.
+
+## Future features
+
+### import.meta.resolve
+
+A dedicated `import.meta.resolve(...)` call is a potential future improvement. It would allow resolving specifiers relatively to the current module in a more straightforward fashion, without extra params:
+
+```js/1/0
+new URL('...', import.meta.url)
+await import.meta.resolve('...')
+```
+
+It would also integrate better with import maps and custom resolvers as it would go through the same module resolution system as `import`. It would also be a stronger signal for bundlers too as it's a static syntax that doesn't depend on runtime APIs like `URL`.
```

Comment:
```suggestion
It would also integrate better with import maps and custom resolvers as it would go through the same module resolution system as `import`. It would be a stronger signal for bundlers too as it's a static syntax that doesn't depend on runtime APIs like `URL`.
```